packages/admin/cms-admin/src/blocks/seo/useSeoTagGeneration.tsx

```diff
@@ -0,0 +1,125 @@
+
+export function useSeoTagGeneration() {
+    const documentContentGenerationApi = useDocumentContentGenerationApi();
+    const errorDialog = useErrorDialog();
+    const apolloClient = useApolloClient();
+
+    const getSeoTag = useCallback(
+        async (content: string): Promise<GQLGenerateSeoTagsMutation["generateSeoTags"]> => {
+            const { data, errors } = await apolloClient.mutate<GQLGenerateSeoTagsMutation, GQLGenerateSeoTagsMutationVariables>({
+                mutation: generateSeoTagsMutation,
+                variables: { content },
+            });
+
+            if (errors || !data?.generateSeoTags) {
+                errorDialog?.showError({
+                    title: (
+                        <FormattedMessage id="comet.blocks.seo.generateSeoTags.errors.failed.title" defaultMessage="Failed to generate SEO tags" />
+                    ),
+                    userMessage: (
+                        <FormattedMessage
+                            id="comet.blocks.seo.generateSeoTags.errors.failed.message"
+                            defaultMessage="An error occurred while generating SEO tags. Please try again."
+                        />
+                    ),
+                    error: errors ? errors[0].message : "Failed to generate SEO tags",
+                });
+                throw new Error("Failed to generate SEO tags");
+            }
+
+            return data.generateSeoTags;
+        },
+        [apolloClient, errorDialog],
+    );
+
+    // During each generateSeoTag request, all tags are generated and cached. The cache makes responses quicker by avoiding unnecessary requests and saves LLM tokens.
+    const generateSeoTag = useCallback(
+        async (tag: SeoTags, currentValue: string | undefined): Promise<string> => {
+            const content = documentContentGenerationApi?.seoBlock?.getDocumentContent().join(" ");
+
+            if (!content || content.length === 0) {
+                errorDialog?.showError({
+                    title: (
+                        <FormattedMessage id="comet.blocks.seo.generateSeoTags.errors.noContent.title" defaultMessage="No content for generation" />
+                    ),
+                    userMessage: (
+                        <FormattedMessage
+                            id="comet.blocks.seo.generateSeoTags.errors.noContent.message"
+                            defaultMessage="This document contains no content that can be used to generate SEO information. Please add content and try again."
+                        />
+                    ),
+                    error: "No content to generate SEO tags from",
+                });
+                throw new Error("No content to generate SEO tags from");
+            }
+
+            const cacheForTagExists = !!seoTagsCache[tag];
+            const currentValueDoesntEqualCachedValue = currentValue !== seoTagsCache[tag];
```

Comment:
What happens if the current value equals the cached value? Will this result in a new attempt?

Reply:
Yes. It should be possible to overwrite a generated value with a new generated value